### PR TITLE
configure the inner client of either BigTableConnection or BigTable

### PR DESCRIPTION
**Pull request by JorritSalverda**
In order to change settings like max_decoding_message_size

Solves https://github.com/liufuyang/bigtable_rs/issues/95

### CODE REVIEW

**Comment by liufuyang** [comment]
@JorritSalverda Thank you for adding this feature. Do you mind rebasing onto current main and I will let the CI build run again? :)

File: bigtable_rs/build.rs [hunk 1]
```diff
@@ -90,7 +90,7 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         )
         .extern_path(".google.protobuf", "::prost_wkt_types")
         .file_descriptor_set_path(&descriptor_file)
-        .compile(
+        .compile_protos(
```
Comment (liufuyang) [inline bigtable_rs/build.rs:93]: Please rebase? The genrated code is also updated, I have made those change in main so you do not need to fix it here.

**Comment by JorritSalverda** [comment]
Hi @liufuyang , it's rebased from your latest changes in main now.

File: bigtable_rs/src/bigtable.rs [hunk 2]
```diff
@@ -403,6 +403,14 @@ impl BigTableConnection {
             timeout: self.timeout.clone(),
         }
     }
+
+    /// Provide a convenient method to update the inner `BigtableClient` so a newly configured client can be set
+    pub fn configure_inner_client(
+        &mut self,
+        config_fn: fn(BigtableClient<AuthSvc>) -> BigtableClient<AuthSvc>,
```
Comment (liufuyang) [inline bigtable_rs/src/bigtable.rs:410]: Oh, one more thing, is it possible to pass in a &mut here, so we do not need to do the clone below?

**Pull request merged**